core/src/test/java/org/elasticsearch/action/bulk/BulkItemResponseTests.java

```diff
@@ -59,23 +55,22 @@ public void testToAndFromXContent() throws IOException {
         for (DocWriteRequest.OpType opType : DocWriteRequest.OpType.values()) {
             int bulkItemId = randomIntBetween(0, 100);
             boolean humanReadable = randomBoolean();
-            BytesReference originalBytes = null;
-            BulkItemResponse expectedBulkItemResponse = null;
 
+            Tuple<? extends DocWriteResponse, ? extends DocWriteResponse> randomDocWriteResponses = null;
             if (opType == DocWriteRequest.OpType.INDEX || opType == DocWriteRequest.OpType.CREATE) {
-                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, randomIndexResponse());
-                originalBytes = toXContent(expectedBulkItemResponse, xContentType, humanReadable);
-
+                randomDocWriteResponses = IndexResponseTests.randomIndexResponse();
             } else if (opType == DocWriteRequest.OpType.DELETE) {
-                expectedBulkItemResponse = new BulkItemResponse(bulkItemId, opType, randomDeleteResponse());
-                originalBytes = toXContent(expectedBulkItemResponse, xContentType, humanReadable);
-
+                randomDocWriteResponses = DeleteResponseTests.randomDeleteResponse();
```

Comment:
Question: later in the DeleteResponseTests you mention in a comment that "the random delete response can contain shard failures with exceptions". Does this matter here?